pyatlan/model/typedef.py

```diff
@@ -233,11 +239,19 @@ def get_valid_values(self) -> Optional[list[str]]:
 
 class AttributeDef(AtlanObject):
     class Options(AtlanObject):
+        custom_metadata_version: str = Field(
+            description="Indicates the version of the custom metadata structure. This determines which other options "
+            "are available and used.",
+            default="v2",
+        )
         description: Optional[str] = Field(
             description="Optional description of the attribute."
         )
         applicable_entity_types: Optional[str] = Field(
-            description="Set of entities on which this attribute can be applied.",
+            description="Set of entities on which this attribute can be applied. "
+            "Note: generally this should be left as-is. Any overrides should instead be applied through "
+            "one or more of applicable_asset_types}, applicable_glossary_types}, or "
+            "applicable_other_asset_types}.",
         )
         custom_applicable_entity_types: Optional[str] = Field(
```

Comment:
Let's at least add to the description of this one that it is deprecated, to give some notice to folks that they shouldn't use this field anymore.

Reply:
Done